### PR TITLE
Update client image to ubuntu 20.04

### PR DESCRIPTION
**Pull request by er-vin**

### CODE REVIEW

File: client/Dockerfile-5.12 [hunk 1]
```diff
@@ -152,3 +125,27 @@ RUN if [ "$BUILD_QT" != "1" ] ; then echo Download precompiled Qt. && \
...
+    cmake ../ && \
+    make -j$(nproc) && \
+    make install && \
+    cd .. && \
+    rm -rf qtkeychain
+
+###########################################################################
+
+# We won't use apt anymore, let's cleanup to shrink the image a bit
+RUN apt-get clean && rm -rf /var/lib/apt/lists/*
```
Comment (MorrisJobke) [inline client/Dockerfile-5.12:150]: That won't work. The cleanup needs to happen in the same step as each step is it's own layer. Thus it was always added at the end of the `apt-get install` step.

Comment (MorrisJobke) [reply client/Dockerfile-5.12:150]: In this case I would just not clean it up or combine all `apt-get` steps 🙈

File: client/Dockerfile-5.12 [hunk 2]
```diff
@@ -152,3 +125,27 @@ RUN if [ "$BUILD_QT" != "1" ] ; then echo Download precompiled Qt. && \
     ; fi
 
 ###########################################################################
+
+ENV QTDIR ${QT_ROOT}
+ENV PATH ${QT_ROOT}/bin:${PATH}
+ENV LD_LIBRARY_PATH ${QT_ROOT}/lib/x86_64-linux-gnu:${QT_ROOT}/lib:/usr/local/lib:${LD_LIBRARY_PATH}
+ENV PKG_CONFIG_PATH ${QT_ROOT}/lib/pkgconfig:${PKG_CONFIG_PATH}
```
[the remark this answers is not in the document]
Comment (er-vin) [reply client/Dockerfile-5.12:132]: > The `ENV`'s here survive for the container run too, right? Did you test this locally @er-vin ?

Yes, and I did this on purpose to simplify the runs down the line.

[the remark this answers is not in the document]
Comment (er-vin) [reply client/Dockerfile-5.12:132]: > Ah, I've just seen the new package at https://download.nextcloud.com/desktop/development/qt/qt-bin-5.12.9-openssl-1.1.1g-linux-x86_64-2020-08-05.tar.xz, so I guess this is the case :)

Yep, obviously had to make that one to test both uses of building and not building Qt with that image. ;-)

File: client/Dockerfile-5.12 (hunk 1 above)
Comment (er-vin) [reply client/Dockerfile-5.12]: Oh indeed, stupid me. Solved in the new replacement commit.

File: client/Dockerfile-5.12 [hunk 3]
```diff
@@ -145,10 +140,28 @@ RUN if [ "$BUILD_QT" = "1" ] ; then \
...
+
+# Install QtKeychain
+RUN cd /tmp && \
+    git clone https://github.com/frankosterfeld/qtkeychain.git && \
+    cd qtkeychain && \
+    git checkout v0.10.0 && \
```
Comment (MorrisJobke) [inline client/Dockerfile-5.12:159]: This does a shallow clone of exactly the commit `v0.10.0` and should save time and traffic. (we do this also on our CI to improve performance)
```suggestion
    git clone https://github.com/frankosterfeld/qtkeychain.git --depth 1 -b v0.10.0 && \
```

**Comment by er-vin** [comment]
Neat, docker hub claims the compressed image size reduced by around 130MB. :-)

**Comment by MorrisJobke** [comment]
```
nextcloudci/client-5.12    client-5.12-10      53a7c28ffa9c        2 hours ago         2.62GB
nextcloudci/client-5.12    client-5.12-9       9e23340e6494        2 weeks ago         2.91GB
```

**Pull request merged**